### ISSUE TITLE
Duplicated RepeatingFragments in Module name 

### ISSUE THREAD

**Issue by byhbt** [nimblehq/compass-credo-plugin#28]
## Bug report

The check `CompassCredoPlugin.Check.RepeatingFragments` is failed for this case `Keyword.Schemas.Keyword` 

![image](https://user-images.githubusercontent.com/948856/209054075-90cde830-db25-45ca-9c83-16a40a72ca24.png)

**Comment by Nihisil** [comment]
I still don't think that it is false positive 🤷‍♂️

**Comment by liamstevens111** [comment]
yeah as I think you mentioned @byhbt , if we want to change this implementation then the check name needs to also change because as per the name it's correct. I think you said "consecutively" or something

**Comment by hanam1ni** [comment]
[The compass mentions](https://nimblehq.co/compass/development/code-conventions/elixir/#convention-39) that the module name should not repeat fragments in a namespace. In the following case, the module name is a repeating fragment of namespaces.
```elixir
Crawler.Keyword.Schemas.Keyword
# Namespace: Crawler.Keyword.Schemas
# Module name: Keyword
```

This is an expected behavior and not considered a false positive in my opinion.

I think we have two options to move forward to solve this issue, feel free to propose additional options.
1. Open RFC for the compass convention about using the plural or singular form for the context module
2. Open RFC for the compass convention to clarify whether should we allow repeating fragments for a non-consecutive namespace or not.

For me I prefer option 1, let me know your thoughts 💭
